File: Data Structures & Algorithms/reorder-linked-list/submission-5.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public void reorderList(ListNode head) {
        ListNode slow = head;
        ListNode fast = head.next;

        while (fast != null && fast.next != null) {
            fast = fast.next.next;
            slow = slow.next;
        }

        ListNode mid = new ListNode();

        mid = slow.next;
        slow.next = null;

        ListNode prev = null;

        while (mid != null) {
            ListNode temp = mid.next;
            mid.next = prev;
            prev = mid;
            mid = temp;
        }

        while (prev != null) {
            ListNode temp1 = head.next;
            ListNode temp2 = prev.next;
            head.next = prev;
            prev.next = temp1; 
            head = temp1;
            prev = temp2;
        }
    }
}

File: Data Structures & Algorithms/rotate-matrix/submission-5.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        m = len(matrix)

        for i in range(m // 2):
            for j in range(i, m - i - 1):
                temp = matrix[i][j]
                matrix[i][j] = matrix[m-1-j][i]
                matrix[m-1-j][i] = matrix[m-1-i][m-1-j]
                matrix[m-1-i][m-1-j] = matrix[j][m-1-i]
                matrix[j][m-1-i] = temp
        return 
        